zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -26,8 +26,8 @@
 import java.util.Set;
 import org.apache.zookeeper.KeeperException;
 import org.apache.zookeeper.WatchedEvent;
-import org.apache.zookeeper.Watcher;
 import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.server.DumbWatcher;
```

Comment:
I agree this is not ideal, the name DumbWatcher sounds hacky. But functionality-wise this is not hacky and it is exactly what we need. The document said it's for bench and unit tests because there are no need for this kind of watcher in production before. We can rename it or change the comment, if we decide to go this way. But let's discuss the approach you proposed first.

Reply:
Synced offline, not going to rename the DumbWatcher class because it's wide usage in the codebase. Will add more comment to explain the reasoning behind using this class.